20.md

```diff
@@ -8,22 +8,42 @@ Web Comments
 
 A special event with kind `34` is defined, meaning "web comment".
 
-Events of this kind should have at least one `r` tag with the value set to the normalized URL of the webpage they're commenting in.
+Events of this kind should have both or at least one of the following tags:
 
-The content should be the plaintext of the comment.
+1- a `r` tag with the value that represents some kind of identifier for the object being commented on. 
+No specific format is enforced, but it preferably should be something that won't change later.  
+A possible format could be: `"<MY WEBSITE NAME> <POST TYPE> <POST ID>"`  
+So when used it'll look something like: `"Github Pull_Request 19"`.  
+This identifier can be placed in a `<meta name='identifier' content='...' />` tag on its page.  
+This will help later if someone wants to create a nostr commenting extension or widget that automatically injects comments on each page.  
+So this way this widget will be able to detect which filter it should be using in the query.  
 
-For example:
+2- a `u` tag with the value set to the normalized URL of the webpage they're commenting in.  
+This is meant to be used in case the identifier wasn't found in the page for whatever reason, so it fallbacks to the page normalized URL.  
 
+### **Why not use only the URL filter?? 🤔**  
+While this might work most of the time, depending only on the URL means that if the URL of the original webpage changes in the future for whatever reason, mean that all the old comments will not appear on the page anymore. So that's why we suggest using another tag that should be more constant.
+
+## Usage
+When you **post** a comment, the object should preferably contain **Both** tags:
 ```json
 {
   "kind": 34,
   "tags": [
-    ["r", "https://random.blog/article"]
+    ["r", "Github Pull_Request 19"],
+    ["u", "https://github.com/nostr-protocol/nips/pull/19"]
```

Comment:
we should just keep the url as `r` and just use nip-14 for the subject.

Reply:
You mean to use the `subject` field to put the identifier that is currently suggested for the `r` tag??